libcontainerd/remote_linux.go

```diff
@@ -351,11 +352,19 @@ func (r *remote) runContainerdDaemon() error {
 		}
 		logrus.Debugf("runContainerdDaemon: runtimeArgs: %s", args)
 	}
+
 	cmd := exec.Command(containerdBinary, args...)
 	// redirect containerd logs to docker logs
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
 	cmd.SysProcAttr = &syscall.SysProcAttr{Setsid: true}
+	cmd.Env = nil
+	// clear the NOTIFY_SOCKET from the env when starting containerd
+	for _, e := range os.Environ() {
+		if !strings.Contains(e, "NOTIFY_SOCKET") {
```

Comment:
@crosbymichael is correct, the env value is used **after** containerd is started in `notifySystem()`